src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Editor/MSBuildXmlAccessor.cs

```diff
@@ -29,12 +29,15 @@ internal class MSBuildXmlAccessor : IProjectXmlAccessor
 
         public async Task<string> GetProjectXmlAsync()
         {
+            var configuredProject = await _unconfiguredProject.GetSuggestedConfiguredProjectAsync().ConfigureAwait(false);
+            if (configuredProject == null)
+            {
+                return string.Empty;
+            }
             using (var access = await _projectLockService.ReadLockAsync())
             {
-                var stringWriter = new EncodingStringWriter(await _unconfiguredProject.GetFileEncodingAsync().ConfigureAwait(true));
-                var projectXml = await access.GetProjectXmlAsync(_unconfiguredProject.FullPath).ConfigureAwait(true);
-                projectXml.Save(stringWriter);
-                return stringWriter.ToString();
+                var projectXml = await access.GetProjectAsync(configuredProject).ConfigureAwait(true);
```

Comment:
`ProjectRootElement.Save` has the side-effect of clearing the dirty flag, which was causing the bug. RawXml does not have this side-effect.